tests/schema/test_resolvers.py

```diff
@@ -540,3 +541,78 @@ def foo(info: Any) -> str:
                 ...
 
         strawberry.Schema(query=Query)
+
+
+class UserLiteral(NamedTuple):
+    id: str
+
+
+def parent_no_self(parent: Parent[UserLiteral]) -> str:
+    return f"User {parent.id}"
+
+
+# Should we allow these?
+def parent_and_self(self, parent: Parent[UserLiteral]) -> str:
+    return f"User {parent.id}"
```

Comment:
I had the same question! I think it might confusing to allow both, so might be nice to throw a warning or error :)

Reply:
Done! I went ahead and also enforced the same for self + root and some other combos under the generic principle of "identify each resource at most once"